packages/build-utils/src/edge-function.ts

```diff
@@ -0,0 +1,61 @@
+import type { File } from './types';
+
+export type EdgeFunctionWasmBinding = {
+  /**
+   * A binding name within the user code.
+   * Will be declared as a global variable.
+   */
+  name: `wasm_${string}`;
+
+  /**
+   * The actual WASM binary to bind.
+   */
+  file: File;
+};
+
+export type EdgeFunction = {
+  type: 'EdgeFunction';
+  /**
+   * A display name for the edge function
+   */
+  name: string;
+
+  /**
+   * The deployment target.
+   * Only v8-worker is currently supported.
+   */
+  deploymentTarget: 'v8-worker';
+
+  /**
+   * The source file. The user code.
+   */
+  script: File;
+
+  /**
+   * The source map for the user code.
+   */
+  scriptSourceMap?: File;
+
+  /**
+   * The environment variables in use for the user code, to be
+   * bound into the edge function
+   */
+  envVarsInUse: string[];
+
+  /**
+   * WebAssembly bindings that the user code requires.
+   */
+  wasmBindings: EdgeFunctionWasmBinding[];
+};
+
+/**
+ * A helper to generate an EdgeFunction
+ */
+export function createEdgeFunction(
```

Comment:
I think we're going to stop using `createLambda()` so I dont see a reason to add a helper for `createEdgeFunction()`.

@TooTallNate thoughts?

Reply:
Agreed.